backends/arm/operators/op_placeholder.py

```diff
@@ -7,167 +7,139 @@
 import serializer.tosa_serializer as ts
 import torch
 from executorch.backends.arm.tosa_mapping import TosaArg
-from executorch.backends.arm.tosa_quant_utils import (
-    get_quant_node_args,
-    is_quant_arg,
-    q_op,
-)
+from executorch.backends.arm.tosa_quant_utils import get_quant_node_args, is_quant_arg
 from executorch.backends.arm.tosa_utils import (
-    is_bias_node_for_addmm,
-    is_consumer_node_depthwise_conv2d,
+    is_bias_node_for_quantized_addmm,
+    is_bias_node_for_quantized_conv,
+    tosa_shape,
 )
 from executorch.exir.dialects._ops import ops as exir_ops
 from torch.export.exported_program import ExportedProgram
 
 
-def process_placeholder(
+def process_inputs(
+    node: torch.fx.Node,
+    tosa_graph: ts.TosaSerializer,
+):
+    """Serialize an input node"""
+    inputs = [TosaArg(node)]
+    input_shape = inputs[0].shape
+    input_dim_order = inputs[0].dim_order
+    tensor = ts.TosaSerializerTensor(
+        inputs[0].name,
+        tosa_shape(input_shape, input_dim_order),
+        ts.DType.INT8 if is_quant_arg(node) else inputs[0].dtype,
+        data=None,
+        placeholderFilename=inputs[0].name + ".npy",
+    )
+    tosa_graph.addInputTensor(tensor)
```

Comment:
IIUC this means we will expect NHWC input tensor, do you want to add a runtime assert for this else it might just consume wrong format data and tricky to debug.